### PR TITLE
DOC-3351: Fixed broken links. Last file fixed: CorDapp upgradeability guarantees.

### PR DESCRIPTION
**Pull request by taylorharvey-r3**

### CODE REVIEW

File: content/en/platform/corda/4.8/enterprise/cordapps/cordapp-build-systems.md [hunk 1]
```diff
@@ -319,7 +319,7 @@ You could sign the CorDapp automatically by:
 
 ### Run development and production modes
 Nodes only accept CorDapps signed by Corda development certificates when running in development mode. If you need to run a CorDapp signed by the (default) development key in the production mode (for example, for testing), add the `cordappSignerKeyFingerprintBlacklist = []` property set to an empty list. See
-[Configuring a node](../node/setup/corda-configuration-file.md#corda-configuration-file-signer-blacklist)).
+[Configuring a node](../../../../../../en/platform/corda/4.8/enterprise/node/setup/corda-configuration-file.html#corda-configuration-file-signer-blacklist)).
```
Comment (michelemarsden) [inline content/en/platform/corda/4.8/enterprise/cordapps/cordapp-build-systems.md:322]: I can't find this header on that page. Closest I got was #limitations.

File: content/en/platform/corda/4.8/enterprise/cordapps/cordapp-constraint-migration.md [hunk 2]
```diff
@@ -42,12 +42,12 @@ For the purposes of constraints migration, it is desirable that any new output s
 
 ### Should I use the **implicit** or **explicit** upgrade path?
 
-The general recommendation for Corda 4 is to use **implicit** upgrades for the reasons described [here](../cordapps/api-contract-constraints.md#implicit-vs-explicit-upgrades). **Implicit** upgrades allow pre-authorising multiple implementations of the contract ahead of time. They do not require additional coding and do not incur a complex choreographed operational upgrade process.
+The general recommendation for Corda 4 is to use **implicit** upgrades for the reasons described [here](../../../../../../en/platform/corda/4.8/enterprise/cordapps/api-contract-constraints.html#implicit-vs-explicit-upgrades). **Implicit** upgrades allow pre-authorising multiple implementations of the contract ahead of time. They do not require additional coding and do not incur a complex choreographed operational upgrade process.
```
Comment (michelemarsden) [inline content/en/platform/corda/4.8/enterprise/cordapps/cordapp-constraint-migration.md:45]: I don't think this title exists anymore: maybe
```suggestion
The general recommendation for Corda 4 is to use **implicit** upgrades for the reasons described [here](../../../../../../en/platform/corda/4.8/enterprise/cordapps/api-contract-constraints.html#implicit-and-explicit-contract-upgrades). **Implicit** upgrades allow pre-authorizing multiple implementations of the contract ahead of time. They do not require additional coding and do not incur a complex choreographed operational upgrade process.
```

File: content/en/platform/corda/4.8/enterprise/cordapps/cordapp-constraint-migration.md [hunk 3]
```diff
@@ -69,7 +69,7 @@ Corda 4.8 requires some additional steps to consume and evolve pre-existing on-l
 
 
 * All Corda Nodes in the same CZ or business network that may encounter a transaction chain with a hash constrained state must be started using
-relaxed hash constraint checking mode as described in [Hash constrained states in private networks](../cordapps/api-contract-constraints.md#hash-constrained-states-in-private-networks).
+relaxed hash constraint checking mode as described in [Hash constrained states in private networks](../../../../../../en/platform/corda/4.8/enterprise/cordapps/api-contract-constraints.html#hash-constrained-states-in-private-networks).
```
Comment (michelemarsden) [inline content/en/platform/corda/4.8/enterprise/cordapps/cordapp-constraint-migration.md:72]: Can't find that header, but can suggest:
```suggestion
relaxed hash constraint checking mode as described in [Hash constrained states in private networks](../../../../../../en/platform/corda/4.8/enterprise/cordapps/api-contract-constraints.html#types-of-contract-constraints).
```

File: content/en/platform/corda/4.8/enterprise/cordapps/cordapp-upgradeability.md [hunk 4]
```diff
@@ -23,18 +23,18 @@ The following guarantees are made for CorDapps running on Corda 4.0
...
+process described in the paragraph below to migrate these to signature constraints. See [CorDapp constraints migration](../../../../../../en/platform/corda/4.8/enterprise/cordapps/cordapp-constraint-migration.md) for more information.
 * CorDapp Contract states generated on ledger using CZ whitelisted constraints are migratable to signature constraints using a manual process
-that requires programmatic code changes. See [CZ whitelisted constraints migration](cordapp-constraint-migration.md#cz-whitelisted-constraint-migration) for more information.
-* Explicit Contract Upgrades are only supported for hash and CZ whitelisted constraint types. See [Performing explicit contract and state upgrades](../upgrading-cordapp.md#performing-explicit-contract-and-state-upgrades) for more information.
+that requires programmatic code changes. See [CZ whitelisted constraints migration](../../../../../../en/platform/corda/4.8/enterprise/cordapps/cordapp-constraint-migration.html#cz-whitelisted-constraint-migration) for more information.
```
Comment (michelemarsden) [inline content/en/platform/corda/4.8/enterprise/cordapps/cordapp-upgradeability.md:32]:
```suggestion
that requires programmatic code changes. See [CZ whitelisted constraints migration](../../../../../../en/platform/corda/4.8/enterprise/cordapps/cordapp-constraint-migration.html#migrating-cz-whitelisted-constraints) for more information.
```

File: content/en/platform/corda/4.8/enterprise/cordapps/upgrading-cordapps.md [hunk 5]
```diff
@@ -330,11 +330,11 @@ removed. This experience will be improved in the future. Making it easier to kil
 There are two types of contract/state upgrade:
 
 
-* *Implicit:* By allowing multiple implementations of the contract ahead of time, using constraints. See [Contract Constraints](api-contract-constraints.md) to learn more.
+* *Implicit:* By allowing multiple implementations of the contract ahead of time, using constraints. See [Contract Constraints](../../../../../../en/platform/corda/4.8/enterprise/cordapps/api-contract-constraints.md) to learn more.
 * *Explicit:* By creating a special *contract upgrade transaction* and getting all participants of a state to sign it using the
 contract upgrade flows.
 
-The general recommendation for Corda 4 is to use **implicit** upgrades for the reasons described [here](api-contract-constraints.md#implicit-vs-explicit-upgrades).
+The general recommendation for Corda 4 is to use **implicit** upgrades for the reasons described [here](../../../../../../en/platform/corda/4.8/enterprise/cordapps/api-contract-constraints.html#implicit-vs-explicit-upgrades).
```
Comment (michelemarsden) [inline content/en/platform/corda/4.8/enterprise/cordapps/upgrading-cordapps.md:337]:
```suggestion
The general recommendation for Corda 4 is to use **implicit** upgrades for the reasons described [here](../../../../../../en/platform/corda/4.8/enterprise/cordapps/api-contract-constraints.html#implicit-and-explicit-contract-upgrades).
```

File: content/en/platform/corda/4.8/enterprise/flow-state-machines.md [hunk 6]
```diff
@@ -662,7 +662,7 @@ loaded off disk again.
 If a node has flows still in a suspended state, with flow continuations written to disk, it will not be
 possible to upgrade that node to a new version of Corda or your app, because flows must be completely “drained”
 before an upgrade can be performed, and must reach a finished state for draining to complete (see
-[Draining the node](upgrading-cordapp.md#draining-the-node) for details). While there are mechanisms for “evolving” serialised data held
+[Draining the node](../../../../../en/platform/corda/4.8/enterprise/upgrading-cordapp.html#draining-the-node) for details). While there are mechanisms for “evolving” serialised data held
```
Comment (michelemarsden) [inline content/en/platform/corda/4.8/enterprise/flow-state-machines.md:665]: Somewhere on this page is an anchor link to #subflows. It should be #sub-flows.

**Review by michelemarsden — COMMENTED** [review]
Few minor changes to anchor links.

File: content/en/platform/corda/4.8/enterprise/cordapps/cordapp-build-systems.md (hunk 1 above)
Comment (taylorharvey-r3) [reply content/en/platform/corda/4.8/enterprise/cordapps/cordapp-build-systems.md:322]: Yeah this was problematic in another link PR too. I don't know if there was originally more to this section or another section entirely, but #limitations does seem to be the best fit.

File: content/en/platform/corda/4.8/enterprise/flow-state-machines.md (hunk 6 above)
Comment (taylorharvey-r3) [reply content/en/platform/corda/4.8/enterprise/flow-state-machines.md:665]: This is also making me realize I have always written subflows, not sub-flows.. Uh oh.

**Review by michelemarsden — APPROVED** [review]
LGTM

**Pull request merged**